addons/account/models/account_journal_dashboard.py

```diff
@@ -174,7 +174,7 @@ def build_graph_data(date, amount, currency):
         result = {}
         for journal in self:
             graph_title, graph_key = journal._graph_title_and_key()
-            currency = journal.currency_id or journal.company_id.currency_id
+            currency = journal.currency_id or journal.company_id.sudo().currency_id
```

Comment:
Note that sudo() has been obfuscated under the variable: developers might be using `currency.<whatever_attribute_or_function>` and not realize they're in sudo mode.

So it's usually a good practice to do something like
```python
journal_currency = journal.currency_id
sudoed_company_currency = journal.company_id.sudo().currency_id  # users might have read access on the journal but not on company
...
data.append(build_graph_data(current_date, random.randint(-5, 15), journal_currency or sudoed_company_currency))
```

not sure sudo**ed** is a thing, though.. but you get the idea :bulb: :grin:

Reply:
Thanks for the feedback. Good point. I think I found a more elegant solution to avoid sudo issues, if that's okay performance-wise.